src/main/java/org/alfresco/repo/domain/schema/script/DeleteNotExistsExecutor.java

```diff
@@ -0,0 +1,555 @@
+ * and/or<br>
+ * <code>deleteNotExistsExecutor.setMaxDeleteBatchSize</code>
+ * <p/>
+ * The statement can be executed in read only mode using: <br>
+ * <code>new DeleteNotExistsExecutor(connection, sql, line, scriptFile, true)</code>
+ * <p/>
+ * The statement 
+ * 
+ * @author Cristian Turlica
+ */
+public class DeleteNotExistsExecutor implements StatementExecutor
+{
+    private static Log logger = LogFactory.getLog(DeleteNotExistsExecutor.class);
+
+    private static final String MSG_EXECUTING_STATEMENT = "schema.update.msg.executing_statement";
+    private static final String ERR_STATEMENT_FAILED = "schema.update.err.statement_failed";
+    private static final String MSG_OPTIONAL_STATEMENT_FAILED = "schema.update.msg.optional_statement_failed";
+
+    public static final String PROPERTY_DEFAULT_BATCH_SIZE = "system.upgrade.default.batchsize";
+    public static final String PROPERTY_DEFAULT_DELETE_BATCH_SIZE = "system.upgrade.default.delete.batchsize";
+
+    private Connection connection;
+    private String sql;
+    private int line;
+    private File scriptFile;
+    private Properties globalProperties;
+    
+    private boolean readOnly = false;
+    private int deleteBatchSize;
+    private int batchSize;
+
+    public DeleteNotExistsExecutor(Connection connection, String sql, int line, File scriptFile, boolean readOnly, Properties globalProperties)
+    {
+        this.connection = connection;
+        this.sql = sql;
+        this.line = line;
+        this.scriptFile = scriptFile;
+        this.readOnly = readOnly;
+        this.globalProperties = globalProperties;
+    }
+
+    public void checkProperties()
+    {
+         PropertyCheck.mandatory(this, "globalProperties", globalProperties);
+    }
+
+    public void execute() throws Exception
+    {
+        checkProperties();
+
+        logger.info("**sql**: " + sql);
+
+        // --DELETE_NOT_EXISTS table.column batch.size.property
+        String[] args = sql.split("[ \\t]+");
+        int sepIndex;
+        if (args.length == 3 && (sepIndex = args[1].indexOf('.')) != -1)
+        {
+            String[] tableColumnArgs = args[1].split(",");
+            if (tableColumnArgs.length >= 2)
+            {
+                // Read the batch size from the named property
+                String batchSizeString = globalProperties.getProperty(args[2]);
+                // Fall back to the default property
+                if (batchSizeString == null)
+                {
+                    batchSizeString = globalProperties.getProperty(PROPERTY_DEFAULT_BATCH_SIZE);
+                }
+
+                batchSize = batchSizeString == null ? 10000 : Integer.parseInt(batchSizeString);
+
+
+                // Read the batch size from the named property
+                String deleteBatchSizeString = globalProperties.getProperty(PROPERTY_DEFAULT_DELETE_BATCH_SIZE);
+                deleteBatchSize = deleteBatchSizeString == null ? 10000 : Integer.parseInt(deleteBatchSizeString);
```

Comment:
10,000 sounds a lot. I would have guessed a smaller number would work well.

Reply:
Added default of 500 but it can be configured to something else via properties.